pkg/gauges/contracts/AuthorizerAdaptor.sol

```diff
@@ -0,0 +1,105 @@
+        _actionIdDisambiguator = bytes32(uint256(address(this)));
+        _vault = vault;
+    }
+
+    /**
+     * @notice Returns the Balancer Vault
+     */
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    /**
+     * @notice Returns the Authorizer
+     */
+    function getAuthorizer() external view returns (IAuthorizer) {
+        return _getAuthorizer();
+    }
+
+    function _getAuthorizer() internal view returns (IAuthorizer) {
+        return getVault().getAuthorizer();
+    }
+
+    /**
+     * @dev As all contracts managed by this adaptor share the same action ID disambiguator
+     *      it's then important to be more granular over `where` an `account` is authorized to act.
+     */
+    function _canPerform(bytes32 actionId, address account, address where) internal view returns (bool) {
+        return _getAuthorizer().canPerform(actionId, account, where);
+    }
+
+    /**
+     * @notice Returns the action ID associated with calling a given function through this adaptor
+     * @dev The contracts managed by this adaptor do not have action ID disambiguators we use the adaptor's globally
+     * @param selector - The 4 byte selector of the function to be called using `performAction`
+     * @return The associated action ID
+     */
+    function getActionId(bytes4 selector) public view override returns (bytes32) {
+        // Each external function is dynamically assigned an action identifier as the hash of the disambiguator and the
+        // function selector. Disambiguation is necessary to avoid potential collisions in the function selectors of
+        // multiple contracts.
+        return keccak256(abi.encodePacked(_actionIdDisambiguator, selector));
+    }
+
+    /**
+     * @notice Performs an arbitrary function call on a target contract, provided the caller is authorized to do so.
+     * @param target - Address of the contract to be called
+     * @param data - Calldata to be sent to the target contract
+     * @return The bytes encoded return value from the performed function call
+     */
+    function performAction(address target, bytes calldata data) external returns (bytes memory) {
+        // We want to check that the caller is authorized to call the function on the target rather than this function.
+        // We must then pull the function selector from `data` rather than `msg.sig`.
+        bytes4 selector;
+        assembly {
+            // The function selector encoded in `data` has an offset relative to the start of msg.data of:
+            // - 4 bytes due to the function selector for `performAction`
+            // - 3 words (3 * 32 = 96 bytes) for `target` and the length and offset of `data`
+            // 96 + 4 = 100 bytes
+            selector := calldataload(100)
+        }
+     
+        _require(_canPerform(getActionId(selector), msg.sender, target), Errors.SENDER_NOT_ALLOWED);
+
+        return target.functionCall(data);
```

Comment:
Recall that we removed `isContract` from `functionCall`, meaning calls to EOAs will succeed here. I think that's fine, but we may want to at least add a comment about it.